earthaccess/auth.py

```diff
@@ -76,6 +76,28 @@ def __init__(self) -> None:
         self.EDL_GENERATE_TOKENS_URL = "https://urs.earthdata.nasa.gov/api/users/token"
         self.EDL_REVOKE_TOKEN = "https://urs.earthdata.nasa.gov/api/users/revoke_token"
 
+    def __repr__(self) -> str:
+        print_str = "Authentication Info\n" + "-------------------\n"
+        for k, v in self.auth_info.items():
+            print_str += str("{}: {}\n".format(k, v))
+
+        return print_str
```

Comment:
I _strongly_ discourage this implementation of `__repr__` for at least 2 reasons:

1. (minor) this is arguably an "abuse" of `__repr__`, which should generally be: "If at all possible, this should look like a valid Python expression that could be used to recreate an object with the same value (given an appropriate environment)." (see https://docs.python.org/3/reference/datamodel.html#object.__repr__)
2. (**very important**) this can lead to inadvertent leaking of the user's token(s) during logging since `auth_info` includes those tokens